code/_onclick/item_attack.dm

```diff
@@ -25,7 +25,17 @@
 	I.attack(src, user)
 
 /mob/living/proc/attacked_by(obj/item/I, mob/living/user, def_zone)
-	apply_damage(I.force, I.damtype, def_zone)
+	var/dmgcheck = apply_damage(I.force, I.damtype, def_zone)
+	if(!dmgcheck)
+		visible_message("<span class='danger'>[user] has attempted to attack [src] with [I].</span>",
+		"<span class='userdanger'>[user] has attempted to attack [src] with [I]!</span>")
+		return 0
+
+	if (I.hitsound && I.force > 0) //If an item's hitsound is defined and the item's force is greater than zero...
+		playsound(get_turf(src), I.hitsound, I.get_clamped_volume(), 1, I.hitsound_extrarange) //...play the item's hitsound at get_clamped_volume() with varying frequency and -1 extra range.
+	else if (I.force == 0)//Otherwise, if the item's force is zero...
```

Comment:
No. This is intended as support for weird healing weapons or whatever. It's part of our code